src/main/java/com/mathworks/ci/RunMatlabTestsBuilder.java

```diff
@@ -247,149 +254,141 @@ private String getInputArguments() {
 
         @Override
-        public void addFilePathArgTo(List<String> inputArgs) {
-            inputArgs.add(MatlabBuilderConstants.JUNIT_RESULTS_PATH + "," + "'"
-                    + getFilePath() + "'");
+        public void addFilePathArgTo(Map<String, String> inputArgs) {
+            inputArgs.put(JUNIT_RESULTS_PATH, getFilePath());
         }
-
     }
 
     public static class CoberturaArtifact extends AbstractArtifactImpl {
 
+        static final String COBERTURA_CODE_COVERAGE_PATH = "CoberturaCodeCoveragePath";
+
         @DataBoundConstructor
         public CoberturaArtifact(String coberturaReportFilePath) {
             super(coberturaReportFilePath);
-         }
-
+        }
 
         @Override
-        public void addFilePathArgTo(List<String> inputArgs) {
-            inputArgs.add(MatlabBuilderConstants.COBERTURA_CODE_COVERAGE_PATH + "," + "'"
-                    + getFilePath() + "'");
+        public void addFilePathArgTo(Map<String, String> inputArgs) {
+            inputArgs.put(COBERTURA_CODE_COVERAGE_PATH, getFilePath());
         }
-
     }
 
     public static class StmResultsArtifact extends AbstractArtifactImpl {
 
+        static final String STM_RESULTS_PATH = "SimulinkTestResultsPath";
+
         @DataBoundConstructor
         public StmResultsArtifact(String stmResultsFilePath) {
             super(stmResultsFilePath);
-         }
-
-
+        }
 
         @Override
-        public void addFilePathArgTo(List<String> inputArgs) {
-            inputArgs.add(MatlabBuilderConstants.STM_RESULTS_PATH + "," + "'"
-                    + getFilePath() + "'");
+        public void addFilePathArgTo(Map<String, String> inputArgs) {
+            inputArgs.put(STM_RESULTS_PATH, getFilePath());
         }
- 
     }
 
     public static class ModelCovArtifact extends AbstractArtifactImpl {
 
+        static final String COBERTURA_MODEL_COVERAGE_PATH = "CoberturaModelCoveragePath";
+
         @DataBoundConstructor
         public ModelCovArtifact(String modelCoverageFilePath) {
             super(modelCoverageFilePath);
-         }
-
+        }
 
         @Override
-        public void addFilePathArgTo(List<String> inputArgs) {
-            inputArgs.add(MatlabBuilderConstants.COBERTURA_MODEL_COVERAGE_PATH + "," + "'"
-                    + getFilePath() + "'");
+        public void addFilePathArgTo(Map<String, String> inputArgs) {
+            inputArgs.put(COBERTURA_MODEL_COVERAGE_PATH, getFilePath());
         }
-
     }
-    
-    public static class NullArtifact implements  Artifact {
+
+    public static class NullArtifact implements Artifact {
 
         @Override
-        public void addFilePathArgTo(List<String> inputArgs) {
+        public void addFilePathArgTo(Map<String, String> inputArgs) {
 
         }
 
         @Override
         public boolean getDefault() {
-            
             return false;
         }
 
         @Override
         public String getFilePath() {
-            
             return null;

```

Comment:
Suggest returning "" instead of null

Reply:
This has to be NULL due to default value in config.jelly  file . UI shows the default value only if the getter of that field is NULL. as we will assign Nullartifact by default after first save operation default values will not be populated on UI as NullArtifact returns  empty string.